ocaml/xapi/repository.ml

```diff
@@ -543,7 +543,24 @@ let get_repository_handler (req : Http.Request.t) s _ =
         match String.sub_to_end req.Request.uri len with
         | uri_path ->
             let root = !Xapi_globs.local_pool_repo_dir in
-            Fileserver.response_file s (Helpers.resolve_uri_path ~root ~uri_path)
+            let uri_path' =
+              match
+                String.starts_with ~prefix:Constants.enabled_repo_uri_prefix
+                  uri_path
+              with
+              | true ->
+                  let repo = get_single_enabled_update_repository ~__context in
+                  let repo_name =
+                    get_remote_repository_name ~__context ~self:repo
+                  in
+                  Str.replace_first
+                    (Str.regexp_string Constants.enabled_repo_uri_prefix)
```

Comment:
We want to avoid the `Str` module; use `Re`. But again, we want to avoid low-level string operations and compose URIs at a higher level.